### ISSUE TITLE
'SSL error: The date of the certificate is invalid' when opening CryptoKitties on Android

### ISSUE THREAD

**Issue by qoqobolo** [status-im/status-react#12073]
# Bug Report

## Problem

<img width="265" alt="Screenshot 2021-04-30 at 16 04 38" src="https://user-images.githubusercontent.com/67952253/116706288-c5643280-a9cd-11eb-9ac8-c87d801184f6.png">

### Reproduction

[comment]: # (Describe how we can replicate the bug step by step.)

- Open Status
- Open Browser tab
- Enter URL cryptokitties.co or navigate to `DApps` > `Collectibles` >` CryptoKitties` > `Open`

### Additional Information

- Status version: Nightly 30/04
- Operating System: Android

**Comment by churik** [comment]
Reproducible on 1.12.1 and on 1.13
Can access dapp via IOS or using Chrome via Metamask, so this behavior is unexpected

**Comment by bitgamma** [comment]
some further notes:

when going to cryptokitties through dap.ps, the cryptokitties loading page displays for a few second and then we get the SSL error.

The address bar changes to

https://xid.cryptokitties.co

I also see the screenshot from @qoqobolo is

https://xid.meetdapper.com

These addresses have indeed expired SSL certificate (same error appears in any browser). So it doesn't look like an actual problem with SSL handling but rather a redirect from the dapp itself. Next step is to investigate why we are being redirected and see if we can fix it from our side, otherwise we'd probably need to contact the dapp owners for more info

**Comment by bitgamma** [comment]
the issue is also reproducible using imToken if typing https://cryptokitties.co manually. If accessing through the app list it works.

**Comment by bitgamma** [comment]
I have sent an email to cryptokitties' support. I don't think this is something we can really fix in status-react.

Debugging in a browser and disabling adblock showed that the wrong url is being loaded via JS by the analytics script. It is actually 3 urls, 2 of which have expired SSL cert and one is invalid domain name. We could maybe do like a desktop browser and just fail only those requests silently but displaying the rest of the content as a workaround.

Seems to be fixable updating to a version of react-native-webview containing https://github.com/react-native-webview/react-native-webview/commit/ef48d35e95ed780bb011e14a7ae18ef110345142#diff-e9700631fc9fa5600d2e1ee70eb2d28aa2a1fbc695b85f99981972d3904ca485

I don't know if there is a reason why we didn't upgrade to latest react-native-webview though

@cammellos wdyt?

**Comment by cammellos** [comment]
@bitgamma 

> We could maybe do like a desktop browser and just fail only those requests silently but displaying the rest of the content as a workaround.

Is definitely the desired behavior. 

We should upgrade, yes, I think we have though some changes in our webview as well that we want to keep https://github.com/status-im/react-native-webview

**Comment by bitgamma** [comment]
@cammellos 

Yeah I didn't notice at first we are using our own branch. I see we added camera/mic permissions prompts. Seems to have been added by recent versions of react-native-webview too however. Was there anything else we added?

I have two options:

1. upgrade to the latest react-native-webview (official branch) and see with the help of QA if there is any regression
2. upgrade our fork to include the needed commit.

Second requires less effort testing because changes would be very limited, but first option frees us from maintaining a fork of that component so I would prefer to do that. Wdyt?
